rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6551,6 +6552,312 @@ private void assertRefundInProcessPegInVersionLegacy(
+            bridgeConstants.getBtcParams(),
+            1,
+            PegTestUtils.createHash(2),
+            merkleRoot,
+            1,
+            1,
+            1,
+            new ArrayList<>()
+        );
+        // simulate blockchain
+        mockChainOfStoredBlocks(
+            btcBlockStore,
+            registerHeader,
+            height + bridgeConstants.getBtc2RskMinimumAcceptableConfirmations(),
+            height
+        );
+
+        InternalTransaction rskTx = new InternalTransaction(
+            Keccak256.ZERO_HASH.getBytes(),
+            0,
+            0,
+            null,
+            null,
+            null,
+            lbcAddress.getBytes(),
+            null,
+            null,
+            null,
+            null
+        );
+
+        co.rsk.core.Coin preCallLbcAddressBalance = repository.getBalance(lbcAddress);
+
+        BigInteger result = bridgeSupport.registerFastBridgeBtcTransaction(
+            rskTx,
+            tx.bitcoinSerialize(),
+            height,
+            pmt.bitcoinSerialize(),
+            derivationArgumentsHash,
+            userRefundBtcAddress,
+            lbcAddress,
+            lpBtcAddress,
+            true
+        );
+
+        if (includeActiveFederation && !includeRetiringFederation) {
+            co.rsk.core.Coin postCallLbcAddressBalance = repository.getBalance(lbcAddress);
+            Assert.assertEquals(
+                preCallLbcAddressBalance.add(co.rsk.core.Coin.fromBitcoin(valueToSend)),
+                postCallLbcAddressBalance
+            );
+
+            verify(provider, times(1)).markFastBridgeFederationDerivationHashAsUsed(
+                tx.getHash(false),
+                fastBridgeDerivationHash
+            );
+
+            verify(provider, times(1)).setFastBridgeFederationInformation(
+                any()
+            );
+        }
+        return result;
+    }
+
+    @Test
+    public void registerFastBridgeBtcTransaction_before_RSKIP293_activation() throws IOException, BlockStoreException, BridgeIllegalArgumentException {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP176)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP219)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(false);
+        Coin valueToSend = Coin.COIN;
+
+        // test sending values to both federations, the active federation and also the retiring federation
+        BigInteger result = registerFastBridgeBtcTransaction_RSKIP293(
+            activations,
+            valueToSend,
+            true,
+            true,
+            true
+        );
+        Assert.assertEquals(co.rsk.core.Coin.fromBitcoin(valueToSend).asBigInteger(), result);
+
+        // test sending values to the active federation, and also to the retiring federation
+        // when there is not currently retiring fed
```

Comment:
Not sure if this is an actual possibility. Send funds to a retiring federation that doesn't exist